src/OrbitGl/TimeGraphLayout.cpp

```diff
@@ -77,6 +78,7 @@ bool TimeGraphLayout::DrawProperties() {
   FLOAT_SLIDER(rounding_num_sides_);
   FLOAT_SLIDER(text_offset_);
   FLOAT_SLIDER(right_margin_);
+  FLOAT_SLIDER(min_button_size_);
```

Comment:
Yeah, the behavior is a bit weird in case this is actually changed during runtime. It depends on how the button is used - if you set the size in every `UpdateLayout` of the parent, then this will work just fine. But if you only set it once, then as you said, buttons will automatically grow, but never shrink. 

TBH I didn't really put any emphasis on fixing this though, these values can only be changed in devmode and aren't meant to be exposed to the user. My main goal was to not allow the button to become unreasonably small because I want to be able to draw a 2-pixel border around it, and I could basically either ignore this case and leave it up to the users of the button, or hard-code it, or just put it into layout.